astropy/utils/metadata.py

```diff
@@ -81,7 +176,18 @@ def merge(left, right, merge_func=concat, metadata_conflicts='warn'):
 
         else:
             try:
-                out[key] = merge_func(left[key], right[key])
+                if merge_func is None:
+                    for left_type, right_type, merge_cls in MERGE_STRATEGY_CLASSES:
+                        if not merge_cls.enabled:
```

Comment:
Problem with this is it enables/disables a merge strategy globally, and sort of leaves it as such.  It's opaque when a merge strategy is being skipped when it's disabled--there's no easy way to check which merge strategies are enabled/disabled.

I'm thinking by default it should be all of them should be disabled, but I'm also thinking about interfaces available though class methods/attributes on the base `MergeStrategy` class to list all merge strategies available.  Maybe if some should be disabled that could be done through an optional argument to `merge()` to exclude strategies.